pipelines/ci.yaml

```diff
@@ -0,0 +1,197 @@
+pool:
+  name: On-Prem Unity
+
+steps:
+- powershell: |
+   # Find unity.exe as Start-UnityEditor currently doesn't support arbitrary parameters
+   $editor = Get-ChildItem ${Env:Unity2018.3.7f1} -Filter 'Unity.exe' -Recurse | Select-Object -First 1 -ExpandProperty FullName
+   
+   $outDir = "$(Build.ArtifactStagingDirectory)\build"
+   $logFile = New-Item -Path "$outDir\build\build_x86.log" -ItemType File -Force
+   
+   $sceneList = "Assets\MixedRealityToolkit.Examples\Demos\HandTracking\Scenes\HandInteractionExamples.unity"
+   
+   $proc = Start-Process -FilePath "$editor" -ArgumentList "-projectPath $(Get-Location) -executeMethod Microsoft.MixedReality.Toolkit.Build.Editor.UnityPlayerBuildTools.StartCommandLineBuild -sceneList $sceneList -logFile $($logFile.FullName) -batchMode -x86 -buildOutput $outDir -buildTarget WSAPlayer -buildAppx" -PassThru
+   $ljob = Start-Job -ScriptBlock { param($log) Get-Content "$log" -Wait } -ArgumentList $logFile.FullName
+   
+   while (-not $proc.HasExited -and $ljob.HasMoreData)
+   {
+       Receive-Job $ljob

```

Comment:
Is it possible to use yaml templates to avoid some of the duplication in these files:

https://github.com/Microsoft/azure-pipelines-agent/issues/1686

i.e. I started out wondering about if we could share like "{Env:Unity2018.3.7f1}" but then noticed there were larger swaths of things that we could possibly template-ize.

Or if this isn't possible, I guess it would be good for our collective knowledge to know if this isn't possible and if so for what reason.

Reply:
Per your other comment, feel free to resolve this as work to do in the future. I didn't understand the scope of what this review was for (i.e. UI -> YAML), where the next step is YAML -> better YAML.